src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -0,0 +1,149 @@
+import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.exceptions.NotFoundException;
+import org.zalando.nakadi.exceptions.TimelineException;
+import org.zalando.nakadi.exceptions.TopicRepositoryException;
+import org.zalando.nakadi.repository.TopicRepository;
+import org.zalando.nakadi.repository.TopicRepositoryHolder;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.repository.db.StorageDbRepository;
+import org.zalando.nakadi.repository.db.TimelineDbRepository;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.view.TimelineRequest;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TimelineService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TimelineService.class);
+    private static final String DEFAULT_STORAGE = "default";
+
+    private final SecuritySettings securitySettings;
+    private final EventTypeCache eventTypeCache;
+    private final StorageDbRepository storageDbRepository;
+    private final TimelineSync timelineSync;
+    private final NakadiSettings nakadiSettings;
+    private final TimelineDbRepository timelineDbRepository;
+    private final TopicRepositoryHolder topicRepositoryHolder;
+
+    @Autowired
+    public TimelineService(final SecuritySettings securitySettings,
+                           final EventTypeCache eventTypeCache,
+                           final StorageDbRepository storageDbRepository,
+                           final TimelineSync timelineSync,
+                           final NakadiSettings nakadiSettings,
+                           final TimelineDbRepository timelineDbRepository,
+                           final TopicRepositoryHolder topicRepositoryHolder) {
+        this.securitySettings = securitySettings;
+        this.eventTypeCache = eventTypeCache;
+        this.storageDbRepository = storageDbRepository;
+        this.timelineSync = timelineSync;
+        this.nakadiSettings = nakadiSettings;
+        this.timelineDbRepository = timelineDbRepository;
+        this.topicRepositoryHolder = topicRepositoryHolder;
+    }
+
+    public void createTimeline(final TimelineRequest timelineRequest, final Client client)
+            throws ForbiddenAccessException, TimelineException, TopicRepositoryException {
+        if (!client.getClientId().equals(securitySettings.getAdminClientId())) {
+            throw new ForbiddenAccessException("Request is forbidden for user " + client.getClientId());
+        }
+
+        try {
+            final String eventTypeName = timelineRequest.getEventType();
+            final EventType eventType = eventTypeCache.getEventType(eventTypeName);
+
+            final Storage storage = storageDbRepository.getStorage(timelineRequest.getStorageId())
+                    .orElseThrow(() -> new NotFoundException("No storage with id: " + timelineRequest.getStorageId()));
```

Comment:
Why have you created some global NotFoundException ?
I think for such situations we use NoSuch...Exception.
Also, with current code that will result in 404 while in other places we usually respond with 422 (e.g. when you create subscription for event-type that is not found)